pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1629,14 +1629,20 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
     if (PyLong_Check(obj)) {
         PRINTMARK();
         tc->type = JT_LONG;
-        GET_TC(tc)->longValue = PyLong_AsLongLong(obj);
-
+        int _overflow = 0;
+        GET_TC(tc)->longValue = PyLong_AsLongLongAndOverflow(obj, &_overflow);
         exc = PyErr_Occurred();
 
+        if (!exc && (_overflow!=0)){
```

Comment:
I could be wrong but I don’t think you need the error check here

Reply:
Nevermind i think this is ok. 

For performance though I think should check that the return value is -1 first before checking for an error, as the latter can be expensive so good to short circuit

Reply:
Ok! Added that. Does seem to work without the error check also though